integration/users/connection_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package users
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/topology"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/internal/util/testutil/testtb"
+)
+
+func TestAuthentication(t *testing.T) {
```

Comment:
Any reason why updating user and creating is tested in connection_test.go instead of in their respective files? :thinking:

Reply:
This is just asking, not requesting code change :see_no_evil:

Reply:
This separate integration test was created to test connection/authentication to the database rather than the create/update methods.

For example, we might use different authentication mechanism combinations in a way that is slightly outside the scope of the create/update logic.

Maybe the checks we do here could live in the existing TestCreateUser and TestUpdateUser tests, but I don't know if that would be better or how to do it tidily.

Any thoughts?

Reply:
I see, let's keep it as it is :+1: . I was trying to differentiate authentication and user related stuff like updating user